From a3d9ef39c0513b6d3b6d6e6c6dab8de268ea59ac Mon Sep 17 00:00:00 2001
From: "gangfeng.xu" <gangfeng.xu@amlogic.com>
Date: Fri, 20 Jun 2014 10:43:20 +0800
Subject: [PATCH 4372/5965] PD #90913 SDIO:fixed sdio init error in autoreboot
 test 20140611/20140616 	the issue log snippt: 	38450 [    7.748337@1]
 init: mount none to target failed 	38451 [    8.247505@1]
 [aml_sdio_print_err] sdio: Timeout error, Cmd5 arg 0x0, 	Xfer 0 Bytes,
 host->xfer_step=12, host->cmd_is_stop=0, pdata->port=0, 	virqs=0x13

---
 drivers/amlogic/mmc/aml_sdhc_m8.c | 8 ++++++--
 drivers/amlogic/mmc/aml_sdio.c    | 6 +++++-
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/mmc/aml_sdhc_m8.c b/drivers/amlogic/mmc/aml_sdhc_m8.c
index 4f5047cba1d2..83ae1148aa74 100755
--- a/drivers/amlogic/mmc/aml_sdhc_m8.c
+++ b/drivers/amlogic/mmc/aml_sdhc_m8.c
@@ -2049,7 +2049,11 @@ static int aml_sdhc_probe(struct platform_device *pdev)
         mmc->f_max = pdata->f_max;
         mmc->max_current_180 = 300; // 300 mA in 1.8V
         mmc->max_current_330 = 300; // 300 mA in 3.3V
-
+        
+        if (aml_card_type_sdio(pdata)) { // if sdio_wifi
+            mmc->rescan_entered = 1; // do NOT run mmc_rescan for the first time
+        }
+        
         if(pdata->port_init)
             pdata->port_init(pdata);
 
@@ -2062,7 +2066,7 @@ static int aml_sdhc_probe(struct platform_device *pdev)
         } else { // ok
             if (aml_card_type_sdio(pdata)) { // if sdio_wifi
                 sdio_host = mmc;
-                mmc->rescan_entered = 1; // do NOT run mmc_rescan for the first time
+                //mmc->rescan_entered = 1; // do NOT run mmc_rescan for the first time
             }
         }
 
diff --git a/drivers/amlogic/mmc/aml_sdio.c b/drivers/amlogic/mmc/aml_sdio.c
index d30cc0ac2bbc..b33845a710a1 100755
--- a/drivers/amlogic/mmc/aml_sdio.c
+++ b/drivers/amlogic/mmc/aml_sdio.c
@@ -1193,6 +1193,10 @@ static int aml_sdio_probe(struct platform_device *pdev)
         mmc->f_min = pdata->f_min;
         mmc->f_max = pdata->f_max;
 
+        if (aml_card_type_sdio(pdata)) { // if sdio_wifi
+            mmc->rescan_entered = 1; // do NOT run mmc_rescan for the first time
+        }
+         
         if(pdata->port_init)
             pdata->port_init(pdata);
 
@@ -1205,7 +1209,7 @@ static int aml_sdio_probe(struct platform_device *pdev)
         } else { // ok
             if (aml_card_type_sdio(pdata)) { // if sdio_wifi
                 sdio_host = mmc;
-                mmc->rescan_entered = 1; // do NOT run mmc_rescan for the first time
+                //mmc->rescan_entered = 1; // do NOT run mmc_rescan for the first time
             }
         }
 
-- 
2.19.0

